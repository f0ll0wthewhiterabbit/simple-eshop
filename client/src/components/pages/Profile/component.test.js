import React from 'react'
import { shallow } from 'enzyme'

import ProfilePage from './component'
import findByTestAttr from '../../../utils/findByTestAttr'

describe('ProfilePage component', () => {
  let wrapper
  const initialProps = {
    firstName: 'John',
    lastName: 'Doe',
  }
  const dataTestAvatar = 'avatar'

  const generateWrapper = passedProps => {
    const defaultProps = { ...initialProps }
    const props = { ...defaultProps, ...passedProps }

    return shallow(<ProfilePage {...props} />)
  }

  beforeEach(() => {
    wrapper = generateWrapper()
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should contain correct abbreviation in avatar', () => {
    const avatar = findByTestAttr(wrapper, dataTestAvatar)

    expect(avatar.text()).toBe('J D')
  })
})
