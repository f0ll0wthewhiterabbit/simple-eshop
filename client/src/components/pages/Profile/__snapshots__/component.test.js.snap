// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage component should render correctly 1`] = `
<WithStyles(ForwardRef(Container))
  component="main"
  maxWidth="xs"
>
  <styled.div>
    <Styled(WithStyles(ForwardRef(Typography)))
      align="center"
      component="h1"
      variant="h4"
    >
      Profile
    </Styled(WithStyles(ForwardRef(Typography)))>
    <Styled(WithStyles(ForwardRef(Avatar)))
      data-test="avatar"
    >
      J D
    </Styled(WithStyles(ForwardRef(Avatar)))>
    <Connect(withRouter(ProfileForm)) />
  </styled.div>
</WithStyles(ForwardRef(Container))>
`;
